.github/actions/deploy-backend/action.yml

```diff
@@ -7,9 +7,28 @@ description: Build and deploy the backend after having installed dfx
 runs:
   using: "composite"
   steps:
+
     - name: Checkout
       uses: actions/checkout@v4
 
+    - name: Restore npm cache
+      uses: actions/cache@v4
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-e2e-npm-${{ hashFiles('package-lock.json') }}
+
+    - name: Restore cargo cache
+      uses: actions/cache@v4
+      with:
+        path: |
+          /home/runner/.cargo/registry
+          /home/runner/.cargo/git
+          target/
+          ~/.cargo/bin/
+        key: ${{ runner.os }}-e2e-cargo-${{ hashFiles('Cargo.lock') }}
```

Comment:
shouldn't the has be including all the rest too?

`hashFiles('Cargo.toml', 'Cargo.lock', 'rust-toolchain.toml', 'src/backend/**/*', 'src/shared/**/*')`

Reply:
as of my knowledge the lock files handle all the dependencies, so if either in npm or cargo builds the dependencies change, we only have to check the lock file.

we can add the other files too, but this could result in more cache files. This could lead to potential issues due to the cache size.

Reply:
for `cargo`, I think we should consider all the changes related to backend: example, we change the code in backend, but cargo stays the same, or if we change some workspace in cargo.toml, etc etc